packages/mcp-provider-code-analyzer/src/actions/run-analyzer.ts

```diff
@@ -110,6 +120,29 @@ export class RunAnalyzerActionImpl implements RunAnalyzerAction {
         const milliseconds: string = String(dateTime.getMilliseconds()).padStart(3, '0');
         return `code-analyzer-results-${year}_${month}_${day}_${hours}_${minutes}_${seconds}_${milliseconds}.json`;
     }
+
+    private emitEngineTelemetry(ruleSelection: RuleSelection, results: RunResults, coreEngineNames: string[]): void {
+        const selectedEngineNames: Set<string> = new Set(ruleSelection.getEngineNames());
+        for (const coreEngineName of coreEngineNames) {
+            if (!selectedEngineNames.has(coreEngineName)) {
+                continue;
+            }
+            if (this.telemetryService) {
+                this.telemetryService.sendEvent(Constants.TelemetryEventName, {
+                    source: Constants.TelemetrySource,
+                    sfcaEvent: Constants.McpTelemetryEvents.ENGINE_SELECTION,
+                    engine: coreEngineName,
+                    ruleCount: ruleSelection.getRulesFor(coreEngineName).length
+                })
+                this.telemetryService.sendEvent(Constants.TelemetryEventName, {
+                    source: Constants.TelemetrySource,
+                    sfcaEvent: Constants.McpTelemetryEvents.ENGINE_EXECUTION,
+                    engine: coreEngineName,
+                    ruleCount: results.getEngineRunResults(coreEngineName).getViolationCount()
```

Comment:
@jfeingold35 I'm comparing the telemetry I'm seeing in the CLI - do we also have access to the violationCount? https://github.com/forcedotcom/code-analyzer/blob/26bfcaa2c27dc4de78725b3a9d1773042549f518/src/lib/actions/RunAction.ts#L115

Reply:
Oh, that's a typo on my part. That property should be `violationCount`. I'll change it.